### PR TITLE
Initial system tests set up for Data Streams Monitoring

### PR DESCRIPTION
**Pull request by hokitam**
## Description

Add initial system tests for Data Streams Monitoring.
1. Create new /dsm endpoint in Java and dotnet. The endpoint triggers Weblog to make a Kafka produce and consume call.
2. Verifies in the agent response that for the `v0.1/pipeline_stats` call, the agent is actually sending the expected data to the backend.
3. Added `test_dsm.py` for both Java and dotnet to confirm expected behavior.

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
4. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
5. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
6. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

File: utils/build/build.sh [hunk 1]
```diff
@@ -188,6 +188,15 @@ build() {
             SYSTEM_TESTS_LIBDDWAF_VERSION=$(docker run --rm system_tests/weblog cat SYSTEM_TESTS_LIBDDWAF_VERSION)
             SYSTEM_TESTS_APPSEC_EVENT_RULES_VERSION=$(docker run --rm system_tests/weblog cat SYSTEM_TESTS_APPSEC_EVENT_RULES_VERSION)
 
+            # Special logic to turn off DD_DATA_STREAMS_ENABLED for spring-boot-native right now
+            # as having it turned on seems to break other existing tests due to a reflection issue
+            # in a library we use.
+            if [ "${WEBLOG_VARIANT}" = "spring-boot-native" ]  || [ "${WEBLOG_VARIANT}" = "spring-boot-3-native" ]; then
```
Comment (cbeauchesne) [inline utils/build/build.sh:194]: if it's related to a given weblog, then define this env directly in the weblog dockerfile. Build.sh must remain generic

File: utils/build/docker/set-system-tests-weblog-env.Dockerfile [hunk 2]
```diff
@@ -14,6 +14,12 @@ ENV DD_TRACE_COMPUTE_STATS=true
 
 ENV SOME_SECRET_ENV=leaked-env-var
 
+# Special setup for DD_DATA_STREAMS_ENABLED
+# until we figure out the issue with enabling
+# data streams on spring-boot-native
+ARG DD_DATA_STREAMS_ENABLED
```
Comment (cbeauchesne) [inline utils/build/docker/set-system-tests-weblog-env.Dockerfile:20]: See comment above (move those line in `spring-boot-native.Dockerfile`

File: utils/interfaces/_agent.py [hunk 3]
```diff
@@ -133,3 +133,6 @@ def get_spans(self, request=None):
                             yield data, span
                         elif get_rid_from_span(span) == rid:
                             yield data, span
+
+    def get_dsm_data(self, request=None):
```
Comment (cbeauchesne) [inline utils/interfaces/_agent.py:137]:
```suggestion
    def get_dsm_data(self):
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Can you include those tests inside `INTEGRATIONS` scenario? features are not overlapping, and every new scenario comes with an high price in term of CI execution (when adding test in existing scenario is almost free).

File: utils/build/docker/set-system-tests-weblog-env.Dockerfile (hunk 2 above)
Comment (hokitam) [reply utils/build/docker/set-system-tests-weblog-env.Dockerfile]: done. I actually only turned on DD_DATA_STREAMS_ENABLED in the dockerfiles where we are testing the configs

File: utils/build/build.sh (hunk 1 above)
Comment (hokitam) [reply utils/build/build.sh]: removed

**Comment by hokitam** [comment]
thanks for the review Charles!, I moved the tests to the INTEGRATIONS scenario as you suggested

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**